### PR TITLE
SW-4501 Refactor the InventoryFilter into FilterMultiSelectContainer

### PR DESCRIPTION
**Pull request by nickgraz**
- so we can easily use the filter multi select elsewhere (planting sites incoming)

### CODE REVIEW

**Comment by nickgraz** [comment]
Current dependencies on/for this PR:
* `main`
  * **PR #1937** <a href="https://app.graphite.dev/github/pr/terraware/terraware-web/1937?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/terraware/terraware-web/1937?utm_source=stack-comment).

File: src/components/common/FilterMultiSelectContainer/index.tsx [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+  filters: T;
+  setFilters: (f: T) => void;
+  label: string;
+  disabled?: boolean;
+  filterKey: keyof T;
+  options: number[];
+  renderOption: (id: number) => string;
+};
+
+export default function FilterMultiSelectContainer<T extends Record<string, number[]>>(
```
Comment (charlie-tf) [inline src/components/common/FilterMultiSelectContainer/index.tsx:57]: Just thinking of whether might need to be `<T extends Record<string, (number | string)[]>>` but honestly, if that's needed we can always add it later

**Review by charlie-tf — APPROVED** [review]
Very nice!

**Comment by nickgraz** [comment]
### Merge activity

* **Dec 20, 1:43 PM**: @@nickgraz started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/terraware/terraware-web/1937).

**Pull request merged**